tools/src/wyvern/tools/typedAST/core/declarations/TypeAbbrevDeclaration.java

```diff
@@ -0,0 +1,186 @@
+
+	public TypeAbbrevDeclaration() {
+		
+	}
+	
+	public TypeAbbrevDeclaration(String alias, Type reference, FileLocation loc) {
+		this.alias = alias;
+		this.reference = reference;
+		this.location = loc;
+				
+	}
+	
+	public Type getReference() {
+		return reference;
+	}
+	
+	public String getAlias() {
+		return alias;
+	}
+
+	@Override
+	public Type getType() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public Map<String, TypedAST> getChildren() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public TypedAST cloneWithChildren(Map<String, TypedAST> newChildren) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public void codegenToIL(GenerationEnvironment environment, ILWriter writer) {
+		// TODO Auto-generated method stub
+		
+	}
+
+	@Override
+	public Expression generateIL(GenContext ctx) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public void writeArgsToTree(TreeWriter writer) {
+		// TODO Auto-generated method stub
+		
+	}
+
+	@Override
+	public FileLocation getLocation() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public Environment extendType(Environment env, Environment against) {
+		// TODO Auto-generated method stub
+		return env;
+	}
+
+	@Override
+	public Environment extendName(Environment env, Environment against) {
+		// TODO Auto-generated method stub
+		return env;
+	}
+
+	@Override
+	public void accept(CoreASTVisitor visitor) {
+		// TODO Auto-generated method stub
+		
+	}
+
+	@Override
+	public String getName() {
+		// TODO Auto-generated method stub
+		return alias;
+	}
+
+	@Override
+	protected Type doTypecheck(Environment env) {
+		if (resolveReferenceType(env) != null) {
+			return new Unit();
+		}
+		else {
+			ToolError.reportError(ErrorMessage.TYPE_NOT_DEFINED, this);
+			return null;
+		}
+		
+	}
+
+	private Type resolveReferenceType(Environment env) {
+		Type resolved_type = TypeResolver.resolve(reference, env);
+		return resolved_type;
+	}
+	
+	@Override
+	protected Environment doExtend(Environment old, Environment against) {
+		return old.extend(new TypeBinding(alias, resolveReferenceType(old)));
+	}
+
+	@Override
+	public EvaluationEnvironment extendWithValue(EvaluationEnvironment old) {
+		// TODO Auto-generated method stub
+		return old;
+	}
+
+	@Override
+	public void evalDecl(EvaluationEnvironment evalEnv,
+			EvaluationEnvironment declEnv) {
+		// TODO Auto-generated method stub
+		
+	}
+
+	@Override
+	public DeclType genILType(GenContext ctx) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public wyvern.target.corewyvernIL.decl.Declaration generateDecl(
+			GenContext ctx, GenContext thisContext) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public wyvern.target.corewyvernIL.decl.Declaration topLevelGen(
+			GenContext ctx) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
```

Comment:
It should be possible to implement topLevelGen and genILType similar to how they are implemented in TypeVarDecl.  A difference is that the internalILType will be a NominalType, not a StructuralType.